pkg/csi/csi.go

```diff
@@ -0,0 +1,243 @@
+
+	Configs aside, this will prepare USB sticks for deploying on baremetal or for recovery and
+	triage.`,
+		PersistentPreRunE: func(c *cobra.Command, args []string) error {
+			return initializeConfig(c)
+		},
+	}
+	c.PersistentFlags().StringVarP(
+		&cfgFile,
+		"config",
+		"c",
+		"",
+		"CSI config file",
+	)
+	_ = c.MarkPersistentFlagFilename("config")
+
+	c.AddCommand(
+		automation.NewCommand(),
+		config.NewCommand(),
+		handoff.NewCommand(),
+		patch.NewCommand(),
+		pit.NewCommand(),
+		DocsCommand(),
+		sls.NewCommand(),
+		version.NewCommand(),
+	)
+	return c
+}
+
+func initializeConfig(c *cobra.Command) error {
+	v := viper.New()
+
+	configFile := c.Flag("config").Value.String()
+	if configFile == "" {
+		// Set the base name of the config file, without the file extension.
+		v.SetConfigName(defaultConfigFilename)
+
+		// Set as many paths as you like where viper should look for the
+		// config file. We are only looking in the current working directory.
+		v.AddConfigPath(".")
+	} else {
+		configFileAbs, _ := filepath.Abs(configFile)
+		v.SetConfigFile(configFileAbs)
+	}
+
+	// Attempt to read the config file, gracefully ignoring errors
+	// caused by a config file not being found. Return an error
+	// if we cannot parse the config file.
+	if err := v.ReadInConfig(); err != nil {
+		// It's okay if there isn't a config file unless one was given.
+		if configFile != "" {
+			return err
+		} else if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			return err
+		}
+	}
+
+	// When we bind flags to environment variables expect that the
+	// environment variables are prefixed, e.g. a flag like --number
+	// binds to an environment variable STING_NUMBER. This helps
+	// avoid conflicts.
+	v.SetEnvPrefix(envPrefix)
+
+	// Environment variables can't have dashes in them, so bind them to their equivalent
+	// keys with underscores, e.g. --favorite-color to STING_FAVORITE_COLOR
+	v.SetEnvKeyReplacer(
+		strings.NewReplacer(
+			"-",
+			"_",
+		),
+	)
+
+	// Bind to environment variables
+	// Works great for simple config names, but needs help for names
+	// like --favorite-color which we fix in the bindFlags function
+	v.AutomaticEnv()
+
+	// Bind the current command's flags to viper
+	bindFlags(
+		c,
+		v,
+	)
+
+	return nil
+}
+
+// Bind each cobra flag to its associated viper configuration (config file and environment variable)
+func bindFlags(cmd *cobra.Command, v *viper.Viper) {
+	cmd.Flags().VisitAll(
+		func(f *pflag.Flag) {
+
+			// Determine the naming convention of the flags when represented in the config file
+			configName := f.Name
+			// If using camelCase in the config file, replace hyphens with a camelCased string.
+			// Since viper does case-insensitive comparisons, we don't need to bother fixing the case, and only need to remove the hyphens.
+			if replaceHyphenWithCamelCase {
+				configName = strings.ReplaceAll(
+					f.Name,
+					"-",
+					"",
+				)
+			}
```

Comment:
Might be worth having a sanitizer function to future proof any other oddly-named ones.  I can unfortunately see that happening.